front/pages/api/w/[wId]/data_sources/index.ts

```diff
@@ -170,6 +171,11 @@ async function handler(
         editedByUserId: user.id,
       });
 
+      const dataSourceType = await getDataSource(auth, ds.name);
```

Comment:
You have `ds` just above which should cover your needs, right?

Reply:
@flvndvd Amplitude code should only use Types and Authenticator, no db model, hence this code.
Was already discussed with spolu, and I agree.

Reply:
I mean we could definitely derive the type from the db model, right?